app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,74 @@
+class OrdersController < ApplicationController
+  before_action :logged_in_user, only: %i(new create index)
+  def new
+    return unless load_book_id_in_cart.empty?
+
+    flash[:danger] = t "order.cart_empty"
+    redirect_to root_path
+  end
+
+  def create
+    list_book = Book.by_book_ids(load_book_id_in_cart)
+    list_shop_id_in_cart.each do |shop_id|
+      order_id = create_order(
+        shop_id, params[:name], params[:address], params[:phone]
+      )
+      add_list_book_to_order_detail list_book, shop_id, order_id
+      update_order_total_price list_book, shop_id, order_id
+    end
+    reset_cart
+    flash[:success] = t "order.success"
+    redirect_to root_path
+  end
+
+  def index; end
+
+  private
+
+  def create_order shop_id, name, address, phone
+    order = current_user.orders.create(
+      shop_id: shop_id,
+      name: name,
+      address: address,
+      phone: phone
+    )
+    order.id
+  end
+
+  def add_list_book_to_order_detail list_book, shop_id, order_id
+    order = load_order_by_id order_id
+    list_book.each do |book|
+      next unless book["shop_id"] == shop_id
+
+      order.order_details.create(
+        quantily: current_cart[book["id"].to_s].to_i,
+        price: book["price"].to_i,
+        book_id: book["id"].to_i
+      )
+    end
+  end
+
+  def update_order_total_price list_book, shop_id, order_id
+    order = load_order_by_id order_id
+    total_bill = 0
+    list_book.each do |book|
+      next unless book["shop_id"] == shop_id
+
+      total_bill += current_cart[book["id"].to_s].to_i * book["price"].to_i
+    end
+    order.update_columns total_price: total_bill
+  end
+
+  def load_order_by_id id
+    Order.find_by id: id
+  end
+
+  def list_shop_id_in_cart
+    books = Book.by_book_ids(load_book_id_in_cart)
+    list_shop_id = []
+    books.each do |book|
+      list_shop_id.push(book.shop_id)
+    end
+    list_shop_id.uniq
```

Comment:
hơi vất vả, sao em ko dùng pluck hay select nhỉ
```suggestion
    list_shop_id = Book.by_book_ids(load_book_id_in_cart).pluck(:shop_id).uniq
```
ri dc hok cu